src/service/xhr-impl.js

```diff
@@ -106,6 +117,18 @@ export class Xhr {
    * @private
    */
   fetch_(input, init) {
+    if (this.ampdoc_) {
+      // TODO(@jridgewell, #9048): We can alternatively _always_ wait on
+      // #whenFirstVisible.
+      if (!viewerForDoc(this.ampdoc_).hasBeenVisible()) {
+        dev().error('XHR', 'attempted to fetch %s before viewer was visible',
```

Comment:
To start with, we should sample it at 1% and then go higher. Otherwise we take a risk of over-spamming our pantheon which is quota-limited. Once confirmed low volume - we can go to 100%.